### ISSUE TITLE
E117: Unknown function: mdip#MarkdownClipboardImage

### ISSUE THREAD

**Issue by stvhwrd** [ferrine/md-img-paste.vim#1]
Hi there, awesome idea for a plugin!  I haven't been able to get it working, yet.  I get this error when I'm in a markdown file, have an image on my clipboard, and use `<leader>p` to paste:

`E117: Unknown function: mdip#MarkdownClipboardImage`

Here's my system info:
```
OS: 64bit Mac OS X 10.13.2 17C205
Kernel: x86_64 Darwin 17.3.0
Uptime: 1d 20h 43m
Packages: 257
Shell: zsh 5.4.2
Resolution: 2880x1800
CPU: Intel Core i7-4770HQ @ 2.20GHz
RAM: 16384MiB
```

Vim version:

```shell
❯❯❯ vim --version

VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jul 26 2017 19:10:24)
Included patches: 1-503, 505-642
Compiled by root@apple.com
Normal version without GUI.  Features included (+) or not (-):
+acl             +file_in_path    -mouse_sgr       +tag_old_static
-arabic          +find_in_path    -mouse_sysmouse  -tag_any_white
+autocmd         +float           -mouse_urxvt     -tcl
-balloon_eval    +folding         +mouse_xterm     -termguicolors
-browse          -footer          +multi_byte      +terminfo
+builtin_terms   +fork()          +multi_lang      +termresponse
+byte_offset     -gettext         -mzscheme        +textobjects
+channel         -hangul_input    +netbeans_intg   +timers
+cindent         +iconv           +num64           +title
-clientserver    +insert_expand   +packages        -toolbar
-clipboard       +job             +path_extra      +user_commands
+cmdline_compl   +jumplist        -perl            +vertsplit
+cmdline_hist    -keymap          +persistent_undo +virtualedit
+cmdline_info    +lambda          +postscript      +visual
+comments        -langmap         +printer         +visualextra
-conceal         +libcall         -profile         +viminfo
+cryptv          +linebreak       +python/dyn      +vreplace
+cscope          +lispindent      -python3         +wildignore
+cursorbind      +listcmds        +quickfix        +wildmenu
+cursorshape     +localmap        +reltime         +windows
+dialog_con      -lua             -rightleft       +writebackup
+diff            +menu            +ruby/dyn        -X11
+digraphs        +mksession       +scrollbind      -xfontset
-dnd             +modify_fname    +signs           -xim
-ebcdic          +mouse           +smartindent     -xpm
-emacs_tags      -mouseshape      +startuptime     -xsmp
+eval            -mouse_dec       +statusline      -xterm_clipboard
+ex_extra        -mouse_gpm       -sun_workshop    -xterm_save
+extra_search    -mouse_jsbterm   +syntax
-farsi           -mouse_netterm   +tag_binary
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X_UNIX  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -L/usr/local/lib -o vim        -lm -lncurses  -liconv -framework Cocoa
```

Here's my `.vimrc`:

```shell
❯❯❯ cat ~/.vimrc

" -------------------------------------------- options ---------------------------------------------
set number              " Show line numbers
set linebreak           " Break lines at word (requires Wrap lines)
set showbreak=+++       " Wrap-broken line prefix
set textwidth=82        " Line wrap (number of cols)
set showmatch           " Highlight matching brace
set visualbell          " Use visual bell (no beeping)

set hlsearch            " Highlight all search results
set smartcase           " Enable smart-case search
set ignorecase          " Always case-insensitive
set incsearch           " Searches for strings incrementally

set autoindent          " Auto-indent new lines
set shiftwidth=4        " Number of auto-indent spaces
set smartindent         " Enable smart-indent
set smarttab            " Enable smart-tabs
set softtabstop=4       " Number of spaces per Tab

" ----------------------------------- variables ----------------------------------
let mapleader = ","     " Remap <leader> key to comma

" ----------------------------------- vim-plug -----------------------------------
if empty(glob('~/.vim/autoload/plug.vim'))
      silent !curl -fLo ~/.vim/autoload/plug.vim --create-dirs
          \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
        autocmd VimEnter * PlugInstall --sync | source $MYVIMRC
endif

" Specify a directory for plugins
" - For Neovim: ~/.local/share/nvim/plugged
" - Avoid using standard Vim directory names like 'plugin'
call plug#begin('~/.vim/plugged')

Plug 'ferrine/md-img-paste.vim'

" Initialize plugin system
call plug#end()

" -------------------------- ferrine's md-img-paste.vim --------------------------
autocmd FileType markdown nmap <silent> <leader>p :call mdip#MarkdownClipboardImage()<CR>
" there are some defaults for image directory and image name, you can change them
" let g:mdip_imgdir = 'img'
" let g:mdip_imgname = 'image'
```

Thanks for any help you might be able to give!  I'm hoping to contribute to this project, as I'll be using it all the time (if I can get it working).

**Comment by ferrine** [comment]
Hi, did you call `:PluginInstall` or related command?

**Comment by ferrine** [comment]
+ cant find these common lines in .vimrc
```
" enable filetype detection:
filetype on
filetype plugin on
filetype indent on " file type based indentation
```

`autocmd FileType markdown ...` works only if markdown is detected

**Comment by stvhwrd** [comment]
Thanks for the effort @ferrine!

* "did you call :PluginInstall or related command?"

  - Yes, I had done this.

* "cant find these common lines in .vimrc:  filetype..."

  - With or without these lines, the plugin would still not work.

I ran `brew install vim` to get a **newer version of vim**, and now it works!  My vimrc is still as posted above.

I upgraded from
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jul 26 2017 19:10:24)
Included patches: 1-503, 505-642
```
to
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jan  8 2018 13:58:05)
Included patches: 1-1400`
```

Thanks for the great plugin!  Cheers 🍻
